Add data report

nikkhilaaryan opened the pull request:
### Summary

Closes #289

- Added `.rst` data report generation via `file_converter.py`
- Redirected stdout to file and versioned it
- Integrated report into Sphinx docs via `index.rst`
- Ensured visibility in rendered HTML
- Fixed mypy module resolution issue with `explicit_package_bases`
- All pre-commit hooks and CI pipelines are passing 

@jannikmi Ready for review

jannikmi commented:
@nikkhilaaryan  thank you.

Please make sure to run the pre-commit hooks.
Note that the Makefiles defines useful commands for this project:
```
pip install uv
make install
make hook
```

Can you please also add the other statictic print statements from the script?

nikkhilaaryan commented:
Ready for review @jannikmi

jannikmi commented:
Thanks, @nikkhilaaryan 

I would suggest a couple of changes:
- use of Contextlib context manager to reroute the stdout to a file instead of opening the file and writing to it with f.write
- deleting an existing file when the file conversion script is being run
- also include the shortcut statistic print statements. The function is being called after the shortcut compilation. 


I can recommend working with a coding agent (e.g. GitHub Copilot)  with a good IDE integration to assist with the implementation. They work great already!

nikkhilaaryan commented:
@jannikmi 
 Implemented all requested changes:

- Replaced manual file writing with `contextlib.redirect_stdout` to cleanly pipe all stats into `data_report.rst`
- Added logic to delete existing report file before regenerating
- Included shortcut statistics using `print_shortcut_statistics()` after `compile_shortcut_mapping()`

All outputs verified -- `data_report.rst` now contains full timezone and shortcut data. 
Ready for re-review

jannikmi commented:
I made some more additions. check out the report: https://timezonefinder.readthedocs.io/en/latest/data_report.html

nikkhilaaryan commented:
Thanks for merging the PR!   @jannikmi 
Everything looks good on the docs.
I'm still early in my open-source, so if you have any suggestions or feedback on how I could improve my contributions, I’d really appreciate it.
